src/github.com/mozilla-services/pushgo/simplepush/handlers_endpoint.go

```diff
@@ -266,34 +280,25 @@ func (h *EndpointHandler) UpdateHandler(resp http.ResponseWriter, req *http.Requ
 	h.metrics.Increment("updates.appserver.incoming")
 
 	// is there a Proprietary Ping for this?
-	if h.pinger == nil {
-		goto sendUpdate
-	}
-	if ok, err = h.pinger.Send(uaid, version, data); err != nil {
+	if updateSent, err = h.doPropPing(uaid, version, data); err != nil {
 		if logWarning {
 			h.logger.Warn("handlers_endpoint", "Could not send proprietary ping",
 				LogFields{"rid": requestID, "uaid": uaid, "error": err.Error()})
 		}
-		goto sendUpdate
-	}
-	/* if this is a GCM connected host, boot vers immediately to GCM
-	 */
-	if ok && h.pinger.CanBypassWebsocket() {
+	} else if updateSent {
```

Comment:
Nit: the style feels wrong here. The updateSent, err should be on a line, then the if err != nil, and else block. Having the updateSent get set in a if pre-condition then used in the else just feels wrong.